src/Initializer/MemoryManager.h

```diff
@@ -357,6 +357,14 @@ class seissol::initializers::MemoryManager {
       ltsParameters = std::make_shared<time_stepping::LtsParameters>(time_stepping::readLtsParametersFromYaml(m_inputParams));
     }
 
+    std::string getOutputPrefix() const {
+      const auto outputDirectory = (*m_inputParams)["output"]["outputfile"];
+      if (!outputDirectory) {
+          logError() << "No outputfile given.";
+      }
+      return outputDirectory.as<std::string>();
+    }
```

Comment:
Could use: https://github.com/SeisSol/SeisSol/blob/731f2b56375e0d7d782bd8064e3da1bc0c3a88a6/src/Initializer/InputAux.hpp#L39 e.g: 
```
const auto outputCandidate = getWithDefault(m_inputParams["output"], "outputfile", "file-not-present");
if (outputCandidate == "file-not-present") {
  logError() << ...
}
return outputCandidate;
```

Reply:
I see your points. To avoid rewriting everything again, I'd favour `getOrFail` in `InputAux.hpp`. Otherwise, parameter reading will become spaghettified too soon.